core/src/test/java/org/bitcoinj/core/FullBlockTestGenerator.java

```diff
@@ -1183,7 +1183,7 @@ public boolean add(Rule element) {
             checkState(b64Original.block.messageSize() == Block.MAX_BLOCK_SIZE);
 
             ByteBuffer buf = ByteBuffer.allocate(b64Original.block.messageSize() + 8);
-            b64Original.block.writeHeader(buf);
+            b64Original.block.asHeader().write(buf);
```

Comment:
Doesn't this change behavior because the copy written to is unreferenced?

Reply:
The target of the write in both cases is `buf`.